command/agent/agent_endpoint.go

```diff
@@ -145,6 +154,135 @@ func (s *HTTPServer) AgentMembersRequest(resp http.ResponseWriter, req *http.Req
 	return out, nil
 }
 
+func (s *HTTPServer) AgentMonitor(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	var secret string
+	s.parseToken(req, &secret)
+
+	// Check agent read permissions
+	if aclObj, err := s.agent.Server().ResolveToken(secret); err != nil {
+		return nil, err
+	} else if aclObj != nil && !aclObj.AllowAgentRead() {
+		return nil, structs.ErrPermissionDenied
+	}
+
+	// Get the provided loglevel.
+	logLevel := req.URL.Query().Get("log-level")
+	if logLevel == "" {
+		logLevel = "INFO"
+	}
+
+	if log.LevelFromString(logLevel) == log.NoLevel {
+		return nil, CodedError(400, fmt.Sprintf("Unknown log level: %s", logLevel))
+	}
+
+	// Determine if we are targeting a server or client
+	nodeID := req.URL.Query().Get("node-id")
+
+	logJSONStr := req.URL.Query().Get("log-json")
```

Comment:
AFAICT we use underscores instead of dashes in query parameters.
```suggestion
	logJSONStr := req.URL.Query().Get("log_json")
```